staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -257,21 +248,12 @@ func AllowProcessTracing(p *corev1.Pod, containerName string, mask ContainerType
 // modifyContainer performs the modifier function m against a container from the
 // the input pod spec which has the name of containerName and the type satisfying
 // the mask.
-func modifyContainer(pod *corev1.Pod, containerName string, mask ContainerType, m func(*corev1.Container)) {
-	if mask&Containers != 0 {
-		for i, c := range pod.Spec.Containers {
-			if c.Name != containerName {
-				continue
-			}
-			m(&pod.Spec.Containers[i])
+func modifyContainer(pod *corev1.Pod, containerName string, mask podutils.ContainerType, m func(*corev1.Container)) {
```

Comment:
In my opinion, there is no need this function. Every caller function can directly pass their Visitor functions(whose interface is `func(c *corev1.Container, _ podutils.ContainerType)`).

Only benefit of this function looks like filtering by `containerName`. I think, we can also do this simply in caller functions.

What do you think @sding3  @verb ?

Reply:
I think it would complicates the call sites, as the all call sites would now have to:
1. perform name filtering,
2. take in a `podutils.ContainerType` in the `VisitFunc` which will be unused since `podutils.VisitContainer` already does type filtering, and 
3. be very careful about whether their `VisitFunc` returns `true` or `false`

Reply:
> I think it would complicate the call sites, as all call sites would now have to:
> 
> 1. perform name filtering,

Name filtering is one line of check
> 2. take in a `podutils.ContainerType` in the `VisitFunc` which will be unused since `podutils.VisitContainer` already does type filtering, and

Isn't the benefit of using `VisitContainer` just does not have to maintain `containerType` in here?
> 3. be very careful about whether their `VisitFunc` returns `true` or `false`

Does `modifyContainer` function handle that case already?

My main concern is we are passing referenced type objects into visitor but these referenced type objects are also being passed to outer `m` function. That makes difficult to follow the execution path if there is any bug occurred in the future.

Reply:
> Name filtering is one line of check

It would actually be 3 lines and the additional mental acrobatics to decide whether to return `true` or `false`:

```go
         if c.Name != containerName {
             // return true or false here?
         }
         // do modification
         // return true or false here?
```

By allowing call sites to decide how they want to filter the container name, we now have also introduce the possibility for another variant where the call site uses:

```go
         if c.Name == containerName {
             // do modification
             // return true or false here?
         }
         // return true or false here?
```

> Does modifyContainer function handle that case already?

Yes, modifyContainer currently handles the returning of the correct boolean value.

With all that said, I'm fine with in-lining all the calls if @verb also request so.

Reply:
I agree with @ardaguclu that this would be better inlined. It's true that it complicates the call site a bit (with return true vs false), but it's a common pattern and also simplifies by shortening the call stack.

Modifying a single container by name would look something like:
```
if c.Name != containerName {
  return true
}

// do modification
return false
```

modifying all containers would be:
```
// do modification
return true
```

Reply:
Sounds good - updated in 1624f5f96dc88e85d5f70f8615685e199d3de93a